include/highfive/bits/H5Easy_scalar.hpp

```diff
@@ -0,0 +1,170 @@
+/*
+ *  Copyright (c), 2017, Adrien Devresse <adrien.devresse@epfl.ch>
+ *
+ *  Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ *
+ */
+#ifndef H5EASY_BITS_SCALAR_HPP
+#define H5EASY_BITS_SCALAR_HPP
+
+#include "../H5Easy.hpp"
+#include "H5Easy_misc.hpp"
+
+namespace HighFive {
+
+namespace detail {
+
+namespace scalar {
```

Comment:
are these xxx_impl methods supposed to be available elsewhere in the project? If not this should be a simple unnamed namespace.

Reply:
I don't understand your question, nor do I understand your suggestion. Sorry.

Reply:
As it is, the entities defined inside this HighFive::detail::scalar namespace are visible outside the "source file" which is know as translation unit in c world. Since I believe they shall not be used anywhere else (they should be involked always from the "public" funcs you define in the bottom) they shoud  be in a special namespace which doesn't create external linkage. Please look up "unnamed namespaces".

Reply:
The idea of unnamed/anonymous namespaces it quite neat, I didn't know. However, this quiet opposes your earlier wish to separate in different files, one for each type. For consistency one would like to also place the functions in `bits/H5Easy_misc.hpp` in an unnamed namespace. In theory that is fine since with all the includes effectively one is creating one translation unit. But I do think that this is not very transparent. For this reason, and for the reason that `bits/H5Easy_scalar.hpp` has to be loaded first, I think that the entire Easy implementation should be in one file (as it originally was).

Reply:
The suggestion would be to use the `details` namespace for internal but shared stuff, which is the case of your `easy_misc`. But again, we can drop the idea, as per the updated comment (i.e.: we are implementing stuff in headers)